File: 0070-climbing-stairs/0070-climbing-stairs.rb
# @param {Integer} n
# @return {Integer}
def climb_stairs(n)
  return n if(n == 1 || n == 2)
  dp = [];
  dp[1] = 1;
  dp[2] = 2;

  for i in 3..n
    dp[i] = dp[i-1] + dp[i-2]
  end

  dp[n]
end

File: 0070-climbing-stairs/0070-climbing-stairs.cs
public class Solution {
    public int ClimbStairs(int n) {
        List<int> steps = new List<int>(){0, 1, 2};
        if(n <= 2){
          return n;
        }
        for(int i = 3; i <= n; i++){
          steps.Add(steps[i-1] + steps[i-2]);
        }
        return steps[n];
    }
}